beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,241 @@
+        yield current_config
+    finally:
+        for process in processes.values():
+            _kill_ganache(process)
+
+
+def _get_chain_map() -> dict[ChainId, int]:
+    number_of_chains = 4
+    return {ChainId(i): 8545 + i for i in range(1, number_of_chains + 1)}
+
+
+def _get_config(
+    chain_map: dict[ChainId, int],
+    contracts,
+    allowance,
+    slave_contract_addresses: dict[ChainId, dict[str, str]],
+) -> Config:
+    account = eth_account.Account.from_key(_LOCAL_ACCOUNT.private_key)
+    url = ape.config.provider.uri
+    deployment_info = {}
+
+    for chain_id in chain_map:
+        contracts_info = dict(
+            RequestManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(
+                    slave_contract_addresses[chain_id]["request_manager"]
+                ),
+                abi=[abi.dict() for abi in contracts.request_manager.contract_type.abi],
+            ),
+            FillManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(slave_contract_addresses[chain_id]["fill_manager"]),
+                abi=[abi.dict() for abi in contracts.fill_manager.contract_type.abi],
+            ),
+        )
+        deployment_info[chain_id] = contracts_info
+    rpc_urls = {"l1": url}
+    for chain_id, port in chain_map.items():
+        rpc_url = f"http://127.0.0.1:{port}"
+        rpc_urls[str(chain_id)] = rpc_url
+
+    token_list = _get_token_list(chain_map, allowance, slave_contract_addresses)
+    config = Config(
+        rpc_urls=rpc_urls,
+        deployment_info=deployment_info,
+        token_checker=TokenChecker(token_list),
+        account=account,
+        fill_wait_time=0,
+        unsafe_fill_time=600,
+        prometheus_metrics_port=None,
+        log_level="debug",
+    )
+    beamer.agent.metrics.init(config=config, source_rpc_url=url, target_rpc_url=url)
+    return config
+
+
+def _start_agent_test(config):
+    agent = Agent(config)
+    agent.start()
+
+    directions = agent.get_directions()
+
+    for direction in directions:
+        with Sleeper(5) as sleeper:
+            while len(agent.get_context(direction).requests) != 1:
+                sleeper.sleep(0.1)
+
+            assert len(agent.get_context(direction).requests) == 1
+
+            request = list(agent.get_context(direction).requests)[0]
+
+            with Sleeper(5) as sleeper:
+                while not request.is_claimed:
+                    sleeper.sleep(0.1)
+    agent.stop()
+
+
+def _get_slave_contract_addresses(proc: psutil.Popen):
+    while True:
+        out = proc.stdout.readline().decode()
+        print(out, flush=True)
+        if "Chain is ready" in out:
+            proc.stdin.write("get_contracts\n".encode())
+            proc.stdin.flush()
+            out = proc.stdout.readline().decode()
+            contract_addresses = json.loads(out.replace("\n", ""))
```

Comment:
Why not instead
```python
line = proc.stdout.readline().decode().strip()
contract_addresses = json.loads(line)
```
?

Since you are reading only one line and only stripping one newline.